utils/git.go

```diff
@@ -435,3 +439,36 @@ func setGoGitCustomClient() {
 	client.InstallProtocol("http", githttp.NewClient(customClient))
 	client.InstallProtocol("https", githttp.NewClient(customClient))
 }
+
+// CreateGitInfoContext Creates GitInfoContest for XSC scans,this is optional.
```

Comment:
```suggestion
// CreateGitInfoContext Creates GitInfoContest for XSC scans, this is optional.
```

Reply:
Too many arguments are passed in this function. Consider passing a struct instead of multiple arguments, or breaking it into smaller functions.